go/libraries/doltcore/sqle/dfunctions/dolt_checkout.go

```diff
@@ -121,10 +122,28 @@ func checkoutRemoteBranch(ctx *sql.Context, dbName string, dbData env.DbData, ro
 		return ErrEmptyBranchName
 	}
 
-	if ref, refExists, err := actions.GetRemoteBranchRef(ctx, dbData.Ddb, branchName); err != nil {
+	if remoteRef, refExists, err := actions.GetRemoteBranchRef(ctx, dbData.Ddb, branchName); err != nil {
 		return errors.New("fatal: unable to read from data repository")
 	} else if refExists {
-		return checkoutNewBranch(ctx, dbName, dbData, roots, branchName, ref.String())
+		err = checkoutNewBranch(ctx, dbName, dbData, roots, branchName, remoteRef.String())
+		if err == nil {
```

Comment:
This is another good place where the code will be slightly cleaner if you deal with the `err != nil` case first, return the error, and then handle the happy path. Nice to get the err handling out of the way immediately and then you won't have the extra level of nesting.